### PR TITLE
Final branch

### PR DESCRIPTION
**Pull request by awais0198**

### CODE REVIEW

File: src/Reducers/index.js [hunk 1]
```diff
@@ -40,6 +41,11 @@ const Reducers = (state = initialState, action) => {
 
       return { ...copyState, sizes: sizes, cart: cart }
 
+    case ADD_MSG:
+      copyState = { ...state }
+      copyState['chat'].push(action.payload.msg)
```
Comment (ahmedwasim7) [inline src/Reducers/index.js]: `copyState.chat.push(action.payload.msg)`

File: src/containers/Chat/index.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import 'containers/Chat/styles.scss'
+
+export const ChatModal = () => {
+  const [isHidden, setState] = useState(false)
+  const [msg, setMsg] = useState('')
+  const chat = useSelector(state => state.chat)
+  const dispatch = useDispatch()
+  const history = useHistory()
+
+  const divClick = () => {
```
Comment (ahmedwasim7) [inline src/containers/Chat/index.js]: Please rename it to `handleClick`

File: src/containers/Chat/index.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    let time = new Date().getHours()
+
+    if (time <= 9 || time >= 20) {
```
Comment (ahmedwasim7) [inline src/containers/Chat/index.js]: Store 9 and 20 in `const`

File: src/containers/Chat/index.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      history.push('/contactus')
+    } else {
+      if (isHidden === true) {
+        setState(false)
+      } else {
+        setState(true)
+      }
```
Comment (ahmedwasim7) [inline src/containers/Chat/index.js]: This can be done in a single line.

File: src/containers/Chat/index.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    }
+  }
+
+  const getChats = () =>
+    chat.map((chat, i) => (
+      <p className={i == 0 ? 'left-msg' : 'right-msg'} key={i}>
```
Comment (ahmedwasim7) [inline src/containers/Chat/index.js]: use `===`

File: src/containers/Chat/index.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        {chat}
+      </p>
+    ))
+
+  const handleChange = e => {
```
Comment (ahmedwasim7) [inline src/containers/Chat/index.js]: Please use destructuring for `target`.

File: src/containers/Chat/styles.scss [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  position: fixed;
+  text-align: center;
+  width: 60px;
+  -webkit-box-shadow: 0 0 15px grey;
+  box-shadow: 0 0 15px grey;
+}
+
+.icon{
+  color: whitesmoke;
+ font-size: 30px;
```
Comment (ahmedwasim7) [inline src/containers/Chat/styles.scss:29]: Spacing issue.

File: src/containers/Chat/styles.scss [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  left: 20px;
+  padding: 20px 8px;
+  position: fixed;
+  text-align: center;
+  width: 60px;
+  -webkit-box-shadow: 0 0 15px grey;
+  box-shadow: 0 0 15px grey;
+}
+
+.icon{
```
Comment (ahmedwasim7) [inline src/containers/Chat/styles.scss]: Spacing issue.

File: src/containers/DetailsDiv/styles.scss [hunk 9]
```diff
@@ -18,3 +18,8 @@
 p, li {
   font-size: 18px;
 }
+
+p,li {
```
Comment (ahmedwasim7) [inline src/containers/DetailsDiv/styles.scss]: Spacing issue.

File: src/containers/DetailsDiv/styles.scss [hunk 10]
```diff
@@ -18,3 +18,8 @@
...
+  font-size: 18px;
+}
+
```
Comment (ahmedwasim7) [inline src/containers/DetailsDiv/styles.scss]: Please remove this empty line.

File: src/containers/index.js [hunk 11]
```diff
@@ -2,3 +2,4 @@ export { DetailsDiv } from 'containers/DetailsDiv'
 export { SelectionDiv } from 'containers/SelectionDiv'
 export { CheckOut } from 'containers/CheckOut'
 export { Home } from 'containers/Home'
+export { ChatModal } from 'containers/Chat'
```
Comment (ahmedwasim7) [inline src/containers/index.js]: Please sort them alphabetically.

**Review by ahmedwasim7 — CHANGES_REQUESTED** [review]

File: src/containers/CheckOut/index.js [hunk 12]
```diff
@@ -10,25 +10,21 @@ export const CheckOut = () => {
...
-          productPrice={price}
-          listId={i + 1}
-        />
-      )
-    }
-    return res
+
+    return cart.map((item, i) => (
+      <CheckOutProduct
+        key={i + 1}
```
Comment (awais0198) [inline src/containers/CheckOut/index.js]: Key kia rkhain then?

File: src/containers/Chat/styles.scss [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+  background-color: #000;
+  border-radius: 10px;
+  color: #fff;
+  cursor: pointer;
+  font-size: 20px;
+  height: 45px;
+  margin: 8px 5px;
+  padding: 5px;
+  width: 60px;
+
```
Comment (ahmedwasim7) [inline src/containers/Chat/styles.scss]: Remove this empty line.

File: src/Components/index.js [hunk 14]
```diff
@@ -0,0 +1,6 @@
+export { ImageSlider } from 'Components/ImageSlider'
+export { Header } from 'Components/Header'
+export { Footer } from 'Components/Footer'
+export { TopHeader } from 'Components/TopHeader'
+export { CheckOutProduct } from 'Components/CheckOutProduct'
+export { ContactUs } from 'Components/ContactUs'
```
Comment (ahmedwasim7) [inline src/Components/index.js]: Please sort them.

File: src/containers/CheckOut/index.js [hunk 15]
```diff
@@ -1,43 +1,53 @@
...
+        productPrice={price}
+        listId={i + 1}
+      />
+    ))
   }
+
+  useEffect(() => {
+    cart.length == 0
+      ? setTotalAmount(0)
+      : setTotalAmount((cart.map(c => c.quantity).reduce((a, b) => a + b) * price) / 100)
```
Comment (ahmedwasim7) [inline src/containers/CheckOut/index.js]:
```
setTotalAmount(((cart.map(c => c.quantity).reduce((a, b) => a + b) || 0) * price) / 100)
```

**Review by ahmedwasim7 — CHANGES_REQUESTED** [review]

**Review by ahmedwasim7 — APPROVED** [review]

**Pull request merged**